### PR TITLE
LF multiple parcels

### PR DESCRIPTION
**Pull request by sithhell**
This patch implements multiple parcel sending in the libfabric parcelport.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,10 +10,10 @@
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 # We require at least CMake V3.0.2
-cmake_minimum_required(VERSION 3.0.2 FATAL_ERROR)
+cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
```
Comment (biddisco) [inline CMakeLists.txt]: Why downgrading cmake from 3.x to 2.x - don't we want to push users to newer versions for better feature support?

File: hpx/runtime/parcelset/parcelhandler.hpp [hunk 2]
```diff
@@ -230,7 +230,7 @@ namespace hpx { namespace parcelset
         ///                 where \a err is the status code of the operation and
         ///                       \a size is the number of successfully
         ///                              transferred bytes.
-        void put_parcels(std::vector<parcel> p, std::vector<write_handler_type> f);
+//         void put_parcels(std::vector<parcel> p, std::vector<write_handler_type> f);
```
Comment (biddisco) [inline hpx/runtime/parcelset/parcelhandler.hpp]: Why so much commented out code? is this to be removed or keeping it for future use?

File: hpx/runtime/parcelset/parcelport_impl.hpp [hunk 3]
```diff
@@ -254,31 +254,35 @@ namespace hpx { namespace parcelset
...
+//                             parcels.size());
+//                     }
+//                     else
+//                     {
+//                         enqueue_parcels(
+//                             dest, std::move(parcels), std::move(handlers));
+//
+//                         get_connection_and_send_parcels(dest);
+//                     }
+//                 })->apply();
```
Comment (biddisco) [inline hpx/runtime/parcelset/parcelport_impl.hpp]: As above. Lot of commented code.

File: plugins/parcelport/libfabric/libfabric_region_provider.hpp [hunk 4]
```diff
@@ -31,7 +34,13 @@ namespace libfabric
...
         template <typename... Args>
         static int register_memory(Args &&... args) {
-            return fi_mr_reg(std::forward<Args>(args)...);
+            int res = fi_mr_reg(std::forward<Args>(args)...);
+            if (res == -FI_ENOKEY)
+                std::cerr << "key already in use\n";
+            if (res == -FI_EKEYREJECTED)
+                std::cerr << "key not available\n";
+
+            return res;
```
Comment (biddisco) [inline plugins/parcelport/libfabric/libfabric_region_provider.hpp]: Have you been getting memory registration errors? I had to reuse the same keys to make things work (with the psm provider) - if this is broken, then I may need to do some bigger changes in the memory handling.

File: src/runtime/parcelset/parcelhandler.cpp [hunk 5]
```diff
@@ -526,162 +526,162 @@ namespace hpx { namespace parcelset
...
+//             resolver_->route(std::move(nonresolved_parcels[i]),
+//                 std::move(nonresolved_handlers[i]));
+//         }
+// #else
+//         HPX_THROW_EXCEPTION(invalid_status,
+//             "Networking was disabled at configuration time. Please "
+//             "reconfigure HPX using -DHPX_WITH_NETWORKING=On.",
+//             "parcelhandler::put_parcels");
+// #endif
+//     }
```
Comment (biddisco) [inline src/runtime/parcelset/parcelhandler.cpp]: ditto commented out code.

**Comment by hkaiser** [comment]
Why does this remove the `put_parcels` function altogether? Could you explain your rationale, please?

**Comment by sithhell** [comment]
No rationale. I accidentally committed too much. Those are mostly my
experiments. I'll revert those changes shortly.

Am 29.05.2017 6:31 nachm. schrieb "Hartmut Kaiser" <notifications@github.com

**Comment by sithhell** [comment]
Sorry about the erronous added changes, all should be clean now.

**Comment by hkaiser** [comment]
I'm going to revert this PR as it seems to be causing reference counting problems...

**Pull request merged**